src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -375,10 +375,19 @@ private boolean isPermitted(AlgorithmConstraints constraints) {
             AlgorithmConstraints constraints,
             List<ProtocolVersion> activeProtocols) {
         List<SignatureScheme> supported = new LinkedList<>();
-        for (SignatureScheme ss: SignatureScheme.values()) {
-            if (!ss.isAvailable ||
-                    (!config.signatureSchemes.isEmpty() &&
-                        !config.signatureSchemes.contains(ss))) {
+
+        // If config.signatureSchemes is non-empty then it means that
+        // it was defined by a System property.  Per
+        // SSLConfiguration.getCustomizedSignatureScheme() the list will
+        // only contain schemes that are in the enum.
+        // Otherwise, use the enum constants (converted to a List).
+        List<SignatureScheme> schemesToCheck =
+                config.signatureSchemes.isEmpty() ?
+                Arrays.asList(SignatureScheme.values()) :
+                config.signatureSchemes;
```

Comment:
I would like to have two more indents for the '?' operator.

Reply:
No problem, will do.